bindings/python/Cargo.toml

```diff
@@ -28,7 +28,7 @@ version = "0.30.3"
 [lib]
 crate-type = ["cdylib"]
 doc = false
-name = "opendal"
+name = "_opendal"
```

Comment:
I think `_opendal` looks worse. Would it be better to use `opendal_python` instead?

I know it's weird to see `opendal.opendal_python`, but `_opendal.so` seems even weirder to me.

Reply:
It's the common practice in Python ecosystem.

Reply:
> For example the std `ssl` module is backed by a `_ssl` extension module.

But they link to `libssl.so` instead of `_ssl.so`, right?

Reply:
Sure, but nobody links to `_opendal`.

Reply:
A `_opendal` python module looks good to me, and it's good to follow the common practice in Python ecosystem. But the so name is really not good...

Reply:
It's just that in Python anything with `_` prefix is considered private, so `opendal_python` would be considered public API which isn't what we want.

Reply:
> It's just that in Python anything with `_` prefix is considered private, so `opendal_python` would be considered public API which isn't what we want.

I understand your point. However, I am concerned about the current setup generating a file named `lib_opendal.so`, which does not seem appropriate to me.

If we have to live with `lib_opendal.so`, I prefer to use `libopendal.so` instead.

Reply:
The final so name in `whl` is actually a little more complicated than you think...

```
Archive:  dist/opendal-0.30.3-cp311-cp311-macosx_11_0_arm64.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
     1740  03-22-2023 03:04   opendal-0.30.3.dist-info/METADATA
      106  03-22-2023 03:04   opendal-0.30.3.dist-info/WHEEL
     2918  03-22-2023 03:04   opendal/__init__.pyi
       81  03-22-2023 03:04   opendal/__init__.py
 49968299  03-22-2023 03:04   opendal/_opendal.cpython-311-darwin.so
      460  03-22-2023 03:04   opendal-0.30.3.dist-info/RECORD
---------                     -------
 49973604                     6 files
```

Reply:
Is it a good idea to use `_opendal_python`?

Reply:
It's fine, just mouthful when you are already in a Python package.

Reply:
FYI, https://github.com/PyO3/maturin/pull/1526 will allow us to just remove `name = "_opendal"` from `[lib]`, as it's already specified in `[package.metadata.maturin]`.

Reply:
That's great!